processing/src/main/java/org/apache/druid/query/filter/DruidPredicateFactory.java

```diff
@@ -32,4 +32,15 @@
   DruidFloatPredicate makeFloatPredicate();
 
   DruidDoublePredicate makeDoublePredicate();
+
+  /**
+   * Object predicate is currently only used by vectorized matchers for non-string object selectors. To preserve
+   * behavior with non-vectorized matchers which use a string predicate with null inputs for these 'nil' matchers,
+   * do the same thing here...
+   */
+  default Predicate<Object> makeObjectPredicate()
```

Comment:
Based on https://github.com/apache/druid/pull/10613#discussion_r578985027 it seems like the intent of this method is that it will only be called for complex types (or maybe nonexistent columns too). If that's the case, please add some javadoc explaining that this method will only be called for complex types. It's analogous to "makeObjectProcessor" in "VectorColumnProcessorFactory". (You could even `@see` to that method in the javadocs if you want.)

Reply:
adjusted javadocs